core/clients.py

```diff
@@ -578,6 +578,7 @@ async def create_log_entry(self, recipient: Member, channel: TextChannel, creato
                 "channel_id": str(channel.id),
                 "guild_id": str(self.bot.guild_id),
                 "bot_id": str(self.bot.user.id),
+                "dm_channel_id": str(recipient.dm_channel.id),
```

Comment:
I believe it's possible for `dm_channel` to be None--more likely with larger servers. As this function gets called before sending the first message to the user, this could potentially be an issue.

You probably want to use `dm_channel = await recipient.create_dm()` beforehand. It returns `dm_channel` if it's cached, or otherwise it will fetch from Discord.